src/App/Expression.cpp

```diff
@@ -517,6 +518,29 @@ Py::Object pyFromQuantity(const Quantity &quantity) {
     }
 }
 
+static const std::regex REGEX_QUANTITY(
+    R"(\s*)"
+    R"(([-+]?[0-9]*[.]?[0-9]+([eE][-+]?[0-9]+)?))" // value
+    R"(\s*)"
+    R"(([^\s]+)?)" // unit
+    R"(\s*)"
+);
+
+// https://github.com/FreeCAD/FreeCAD/issues/11825
+static Quantity parseQuantityFromText(std::string text) {
```

Comment:
Are you familiar with the Google Test framework? I'd love to see a couple of unit tests targeting this method.